dex/farm/src/base_functions.rs

```diff
@@ -204,20 +206,68 @@ pub trait BaseFunctionsModule:
         reward
     }
 
-    fn migrate_old_farm_positions(&self, caller: &ManagedAddress) {
+    fn migrate_old_farm_positions(&self, caller: &ManagedAddress) -> BigUint {
         let payments = self.get_non_empty_payments();
         let farm_token_mapper = self.farm_token();
         let farm_token_id = farm_token_mapper.get_token_id();
+        let mut migrated_amount = BigUint::zero();
         for farm_position in &payments {
             if farm_position.token_identifier == farm_token_id
                 && self.is_old_farm_position(farm_position.token_nonce)
             {
-                let mut user_total_farm_position = self.get_user_total_farm_position(caller);
-                user_total_farm_position.total_farm_position += farm_position.amount;
-                self.user_total_farm_position(caller)
-                    .set(user_total_farm_position);
+                migrated_amount += farm_position.amount;
             }
         }
+
+        if migrated_amount > 0 {
+            let mut user_total_farm_position = self.get_user_total_farm_position(caller);
+            user_total_farm_position.total_farm_position += &migrated_amount;
+            self.user_total_farm_position(caller)
+                .set(user_total_farm_position);
+        }
+
+        migrated_amount
+    }
+
+    fn decrease_old_farm_positions(&self, migrated_amount: BigUint, caller: &ManagedAddress) {
+        if migrated_amount == BigUint::zero() {
+            return;
+        }
+        self.user_total_farm_position(caller)
+            .update(|user_total_farm_position| {
+                user_total_farm_position.total_farm_position -= migrated_amount;
+            });
+    }
+
+    fn try_set_farm_position_migration_nonce(&self) {
+        if !self.farm_position_migration_nonce().is_empty() {
+            return;
+        }
+
+        let farm_token_mapper = self.farm_token();
+
+        let attributes = FarmTokenAttributes {
+            reward_per_share: BigUint::zero(),
+            entering_epoch: 0,
+            compounded_reward: BigUint::zero(),
+            current_farm_amount: BigUint::zero(),
+            original_owner: self.blockchain().get_sc_address(),
+        };
+
+        let migration_farm_token_nonce = if farm_token_mapper.get_token_state().is_set() {
```

Comment:
no need to create new token. Use VM endpoint GetCurrentESDTNFTNonce / implement it to Rust Framework if it does not exist.

Reply:
Updated.